internal/webhook/package_validator.go

```diff
@@ -119,6 +119,13 @@ func (p *PackageValidatingWebhook) validateCreateOrUpdate(ctx context.Context, p
 		return err
 	}
 
+	// Scope validation
+	if (manifest.Scope == nil || *manifest.Scope == "Cluster") && pkg.IsNamespaceScoped() {
+		return errors.New("invalid scope, expected ClusterPackage but found Package")
+	} else if (manifest.Scope != nil && *manifest.Scope == "Namespaced") && !pkg.IsNamespaceScoped() {
+		return errors.New("invalid scope, expected Package but found ClusterPackage")
+	}
```

Comment:
Please use `manifest.Scope.IsCluster()` and `manifest.Scope.IsNamespaced()`. These already include the necessary nil checks.

Reply:
Sure, I realized the same, yesterday after making the PR. I have now used the pre-existing checks in PackageScope.